knowledge_repo/app/index.py

```diff
@@ -29,6 +34,8 @@ def update_index():
             post.status = current_repo.PostStatus.UNPUBLISHED
             continue
 
+        post.keywords = get_keywords(post)
```

Comment:
@matthewwardrop one issue with running it only in the update_metadata_from_kp is that tag changing won't trigger a revision update. @NiharikaRay wonder if there's a place to call this for tag updating also?